homeassistant/components/streamlabswater/config_flow.py

```diff
@@ -0,0 +1,131 @@
+"""Config flow for StreamLabs integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from streamlabswater.streamlabswater import StreamlabsClient
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import DOMAIN as HOMEASSISTANT_DOMAIN, HomeAssistant
+from homeassistant.data_entry_flow import AbortFlow, FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+
+from .const import DOMAIN, LOGGER
+
+ISSUE_PLACEHOLDER = {"url": "/config/integrations/dashboard/add?domain=streamlabswater"}
+
+
+async def validate_input(hass: HomeAssistant, api_key: str) -> None:
+    """Validate the user input allows us to connect."""
+    client = StreamlabsClient(api_key)
+    response = await hass.async_add_executor_job(client.get_locations)
+    locations = response.get("locations")
+
+    if locations is None:
+        raise CannotConnect
+
+
+class StreamlabsConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for StreamLabs."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            self._async_abort_entries_match(user_input)
```

Comment:
Unique id by the api_key?

Reply:
That's the only info I know we have, for now when importing its enough imo

Reply:
Was meaning if we should use the api_key as unique id instead but I guess same same

Reply:
unique_id sounds so official